python/ray/serve/controller.py

```diff
@@ -735,96 +677,33 @@ def _checkpoint(self) -> None:
-        internal state will be updated and long-poll clients may be notified.
-
-        Performs the following operations:
-            1) Deserializes the internal state from the checkpoint.
-            2) Starts/stops any replicas that are pending creation or
-               deletion.
-        """
-        start = time.time()
-        logger.info("Recovering from checkpoint")
-
-        self.actor_reconciler = checkpoint.reconciler
-
-        self._serializable_inflight_results = checkpoint.inflight_reqs
-        for uuid, fut_result in self._serializable_inflight_results.items():
-            self._create_event_with_result(fut_result.requested_goal, uuid)
-
-        # NOTE(edoakes): unfortunately, we can't completely recover from a
-        # checkpoint in the constructor because we block while waiting for
-        # other actors to start up, and those actors fetch soft state from
-        # this actor. Because no other tasks will start executing until after
-        # the constructor finishes, if we were to run this logic in the
-        # constructor it could lead to deadlock between this actor and a child.
-        # However, we do need to guarantee that we have fully recovered from a
-        # checkpoint before any other state-changing calls run. We address this
-        # by acquiring the write_lock and then posting the task to recover from
-        # a checkpoint to the event loop. Other state-changing calls acquire
-        # this lock and will be blocked until recovering from the checkpoint
-        # finishes. This can be removed once we move to the async control loop.
-
-        async def finish_recover_from_checkpoint():
-            assert self.write_lock.locked()
-            self.autoscaling_policies = await self.actor_reconciler.\
-                _recover_from_checkpoint(self.backend_state, self)
-            self.write_lock.release()
-            logger.info(
-                "Recovered from checkpoint in {:.3f}s".format(time.time() -
-                                                              start))
-
-        await self.write_lock.acquire()
-        asyncio.get_event_loop().create_task(finish_recover_from_checkpoint())
-
-    async def do_autoscale(self) -> None:
-        for backend, info in self.backend_state.backends.items():
-            if backend not in self.autoscaling_policies:
-                continue
-
-            new_num_replicas = self.autoscaling_policies[backend].scale(
-                self.backend_stats[backend], info.backend_config.num_replicas)
-            if new_num_replicas > 0:
-                await self.update_backend_config(
-                    backend, BackendConfig(num_replicas=new_num_replicas))
-
     async def reconcile_current_and_goal_backends(self):
         pass
 
     def set_goal_id(self, goal_id: UUID) -> None:
         event = self.inflight_results.get(goal_id)
-        logger.debug(f"Setting Goal Id: {goal_id}")
```

Comment:
Is it okay to keep this?

Reply:
Sure. I found it very verbose, might remove it in the future if it isn't useful